session/pingpong/consumer_totals_storage.go

```diff
@@ -85,6 +87,47 @@ func (cts *ConsumerTotalsStorage) Get(chainID int64, id identity.Identity, herme
 	return res, err
 }
 
+// Add adds the given amount as promised for the given channel, if none exists it adds from 0.
+func (cts *ConsumerTotalsStorage) Add(chainID int64, id identity.Identity, hermesID common.Address, amount *big.Int) error {
+	cts.lock.Lock()
+	defer cts.lock.Unlock()
+
+	key := cts.makeKey(chainID, id, hermesID)
+	value := new(big.Int)
+	err := cts.bolt.GetValue(consumerTotalStorageBucketName, key, &value)
+	if err != nil {
+		// wrap the error to an error we can check for
+		if errors.Is(storm.ErrNotFound, err) {
+			log.Debug().Msg("No previous invoice grand total, assuming zero")
+			value = big.NewInt(0)
+		} else {
+			return fmt.Errorf("could not get total promised to add : %w", err)
+		}
+	}
+	if value == nil || value.Cmp(big.NewInt(0)) == -1 {
```

Comment:
Can it even be `nil`? https://go.dev/play/p/I5VFOAy5_kD

Reply:
fun fact: [yes](https://go.dev/play/p/_PH7DtQ2jed)